skyrl-train/examples/mini_swe_agent/mini_swe_trainer.py

```diff
@@ -0,0 +1,72 @@
+from typing import List
+import torch
+import math
+from loguru import logger
+
+from skyrl_train.generators.base import (
+    GeneratorInput,
+    GeneratorOutput,
+)
+from skyrl_train.trainer import RayPPOTrainer
+
+
+class MiniSWEPPOTrainer(RayPPOTrainer):
+    @torch.no_grad()
+    async def generate(
+        self,
+        input_batch: GeneratorInput,
+    ) -> GeneratorOutput:
+        """
+        Generate rollouts.
+
+        If colocate_all is enabled:
+        - before calling this method, the policy model should be on CPU and inference engine should
+            be awake (i.e. on GPU).
+        - after calling this method, the same model placement still holds.
+        """
+        generator_output: GeneratorOutput = await self.generator.generate(input_batch)
+
+        # add rollout metrics to self.all_metrics
+        if generator_output["rollout_metrics"] is not None:
+            self.all_metrics.update(generator_output["rollout_metrics"])
+
+        # NOTE (sumanthrh): We filter out instances that failed during generation, so the number of prompts at input can differ from number of trajectories at output
+        # we simply ignore validation for now
+        # validate_generator_output(input_batch, generator_output)
+
+        logger.info(f"Rewards: {generator_output["rewards"]}")
+
+        return generator_output
+
+    def postprocess_generator_output(self, generator_output: GeneratorOutput, uids: List[str]) -> GeneratorOutput:
+        # first: add custom validation
```

Comment:
This whole set of code in this method definitely shouldn't be in this PR, why is it needed? It makes me believe we currently don't quite have the right abstractions.